bot/exts/utilities/bookmark.py

```diff
@@ -152,6 +151,34 @@ def event_check(reaction: discord.Reaction, user: discord.Member) -> bool:
 
         await reaction_message.delete()
 
+    @bookmark.command(name="delete", aliases=("del", "rm"), root_aliases=("unbm", "unbookmark", "dmdelete", "dmdel"))
+    @whitelist_override(bypass_defaults=True, allow_dm=True)
+    async def delete_bookmark(
+        self,
+        ctx: commands.Context,
+        target_message: Optional[WrappedMessageConverter]
+    ) -> None:
+        """
+        Delete the target message specified by the member. Must be ran in DMs targetting a message sent by Sir-Lancebot.
+
+        `target_message` may be a message link, or an ID that references the message,
+        Instead of giving a `target_message` the user may also reply to the message to delete.
+
+        This command allows deleting any message sent by Sir-Lancebot in the user's DMs.
+        """
+        target_message: Optional[discord.Message] = target_message or getattr(ctx.message.reference, "resolved", None)
+        if target_message is None:
+            raise commands.UserInputError(MESSAGE_NOT_FOUND_ERROR)
+
+        if not isinstance(ctx.channel, discord.DMChannel):
+            raise commands.UserInputError("You can only run this command your own DMs!")
```

Comment:
Is this actually possible with the whitelist above? If the purpose is for type-casting, we can be more explicit about that.

Reply:
Yea, it is somehow.
![image](https://user-images.githubusercontent.com/9753350/185588840-6a80f7a6-da84-460a-be48-548449a41d2f.png)

Reply:
I think this is something worth investigating as a bug in the whitelist code, or maybe d.py, but I won’t hold up this PR on it. If you have reproduction steps that would be great.

Reply:
Agreed, I might hold off until after I bump lance up to stable d.py 2.0. Just in the hopes that it disappears.